keda/src/components/common/CommonComponents.tsx

```diff
@@ -0,0 +1,607 @@
+        <CircularProgress />
+      </Box>
+    );
+  }
+
+  return (
+    <Box display="flex" justifyContent="center" alignItems="center" p={2} minHeight="200px">
+      <Grid container spacing={2} direction="column" justifyContent="center" alignItems="center">
+        <Grid item>
+          <Typography variant="h5">
+            KEDA was not detected on your cluster. If you haven't already, please install it.
+          </Typography>
+        </Grid>
+        <Grid item>
+          <Typography>
+            Learn how to{' '}
+            <MuiLink
+              href="https://keda.sh/docs/latest/deploy"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              install
+            </MuiLink>{' '}
+            KEDA
+          </Typography>
+        </Grid>
+      </Grid>
+    </Box>
+  );
+}
+
+interface BaseKedaAuthenticationProps {
+  title?: string;
+  resourceType: typeof TriggerAuthentication | typeof ClusterTriggerAuthentication;
+}
+
+export function BaseKedaAuthenticationDetail({ resourceType }: BaseKedaAuthenticationProps) {
+  const { name, namespace } = useParams<{ name: string; namespace?: string }>();
+  const { isKedaInstalled, isKedaCheckLoading } = useKedaInstalled();
+
+  const renderValue = (value: any): ReactNode => {
+    if (Array.isArray(value)) {
+      if (value.every(item => typeof item === 'object' && item !== null)) {
+        return value.map((item, index) => (
+          <NameValueTable
+            key={index}
+            rows={Object.entries(item).map(([k, v]) => ({
+              name: k,
+              value: renderValue(v),
+            }))}
+          />
+        ));
+      }
+    }
+
+    if (typeof value === 'object' && value !== null) {
+      return (
+        <NameValueTable
+          rows={Object.entries(value).map(([k, v]) => ({
+            name: k,
+            value: renderValue(v),
+          }))}
+        />
+      );
+    }
+
+    return String(value);
+  };
+
+  return (
+    <>
+      {isKedaInstalled ? (
+        <DetailsGrid
+          resourceType={resourceType}
+          name={name}
+          withEvents
+          namespace={resourceType.isNamespaced ? namespace : undefined}
+          extraInfo={item =>
+            item && [
+              { name: 'API Version', value: resourceType.apiVersion },
+              { name: 'Kind', value: resourceType.kind },
+            ]
+          }
+          extraSections={item =>
+            item && [
+              item.spec &&
+                Object.keys(item.spec).length > 0 && {
+                  id: 'spec',
+                  section: (
+                    <SectionBox title="Spec">
+                      {Object.entries(item.spec).map(([key, value]) => (
+                        <Fragment key={key}>
+                          <h2>{key}</h2>
+                          {renderValue(value)}
+                        </Fragment>
+                      ))}
+                    </SectionBox>
```

Comment:
The secretTargetRef is displayed in the spec and we should have a way to jump to the secrets detail view from here.